tsl/test/isolation/specs/freeze_chunk.spec

```diff
@@ -72,17 +73,43 @@ step "comp_s2_compress"
 	{ BEGIN; SELECT CASE WHEN compress_chunk(ch) IS NOT NULL THEN 'Success' ELSE 'Failed' END  as COL FROM (SELECT show_chunks('measurements') ch ) q; }
 step "comp_s2_commit" { COMMIT; }
 
-###freeze_chunk waits for pending insert txns to complete
+session "decomp_s2"
+step "decomp_s2_decompress" 
+	{ BEGIN; SELECT CASE WHEN decompress_chunk(ch) IS NOT NULL THEN 'Success' ELSE 'Failed' END  as COL FROM (SELECT show_chunks('measurements') ch ) q; }
+step "decomp_s2_commit" { COMMIT; }
+
+#### freeze_chunk ####
+
+###freeze_chunk waits for pending insert txns to complete, uncompressed and compressed variant
 permutation "s2_wp_enable" "ins_s2_insert" "s2_wp_release" "s1_freeze" "ins_s2_commit" "ins_s2_query" "s1_status"
+permutation "comp_s2_compress" "comp_s2_commit" "s2_wp_enable" "ins_s2_insert" "s2_wp_release" "s1_freeze" "ins_s2_commit" "ins_s2_query" "s1_status"
 
-###freeze_chunk waits for pending update txns to complete
+###freeze_chunk waits for pending update txns to complete, uncompressed and compressed variant
 permutation "upd_s2_wp_enable" "upd_s2_update" "upd_s2_wp_release" "s1_freeze" "upd_s2_commit" "upd_s2_query" "s1_status"
+permutation "comp_s2_compress" "comp_s2_commit" "upd_s2_wp_enable" "upd_s2_update" "upd_s2_wp_release" "s1_freeze" "upd_s2_commit" "upd_s2_query" "s1_status"
 
-###freeze_chunk waits for pending delete txns to complete
+###freeze_chunk waits for pending delete txns to complete, uncompressed and compressed variant
 permutation "del_s2_wp_enable" "del_s2_delete" "del_s2_wp_release" "s1_freeze" "del_s2_commit" "del_s2_query" "s1_status"
+permutation "comp_s2_compress" "comp_s2_commit" "del_s2_wp_enable" "del_s2_delete" "del_s2_wp_release" "s1_freeze" "del_s2_commit" "del_s2_query" "s1_status"
 
-###freeze_chunk and select do not block each other
+###freeze_chunk and select do not block each other, uncompressed and compressed variant
 permutation "sel_s2_wp_enable" "sel_s2_query" "sel_s2_wp_release" "s1_freeze" "sel_s2_commit" "s1_status"
+permutation "comp_s2_compress" "comp_s2_commit" "sel_s2_wp_enable" "sel_s2_query" "sel_s2_wp_release" "s1_freeze" "sel_s2_commit" "s1_status"
 
 ##if compress_chunk is in progress, freeze_chunk is blocked
 permutation "LockChunk1" "comp_s2_compress" "s1_freeze" "UnlockChunk" "comp_s2_commit" "s1_status" 
+
+##if decompress_chunk is in progress, freeze_chunk is blocked
+permutation "comp_s2_compress" "comp_s2_commit" "LockChunk1" "decomp_s2_decompress" "s1_freeze" "UnlockChunk" "decomp_s2_commit" "s1_status" 
+
+#### unfreeze_chunk ####
+
+###unfreeze_chunk and select do not block each other, uncompressed and compressed variant
+permutation "s1_freeze" "sel_s2_wp_enable" "sel_s2_query" "sel_s2_wp_release" "s1_unfreeze" "sel_s2_commit" "s1_status"
+permutation "comp_s2_compress" "comp_s2_commit" "s1_freeze" "sel_s2_wp_enable" "sel_s2_query" "sel_s2_wp_release" "s1_unfreeze" "sel_s2_commit" "s1_status"
+
+##if compress_chunk is in progress, unfreeze_chunk is blocked
```

Comment:
This should not happen because a frozen chunk cannot be compressed?
We can only unfreeze a frozen chunk?
are you checking concurrent status updates here (if so, please add a comment).

Reply:
would be good to have a freeze  + unfreeze test.However, I don't think we can get deterministic output for it. So that will probably not work for CI/CD  without some debug waitpoints.